File: python/sin.py
import matplotlib.pyplot as plt
import numpy as np

# Initialize the figure and axis for plotting
fig, ax = plt.subplots()

# Main drawing logic
for Y in np.arange(-10, 10, 0.2):
    for X in np.arange(-10, 10, 0.3):
        R = np.sqrt(X**2 + Y**2) * 1.5
        F = 90
        if R > 0:
            F = abs(90 * np.sin(R) / R)
        A = int(10 * X + 160 - 5 * Y)
        B = int(5 * Y + 2.5 * X + 93)
        ax.plot(A, B - F, 'o', color='black', markersize=1)

# Set the background color to black
#ax.set_facecolor('black')

# Set the aspect of the plot to be equal
#ax.set_aspect('equal')

# Remove the axes for a cleaner look
#plt.axis('off')

# Show the plot
plt.show()

File: p5/sin.js
function setup() {
  createCanvas(400, 400);
}

function draw() {
  background(50);
  stroke(255, 165, 0);
  strokeWeight(2);

	// Main drawing logic
	for (let y = -10; y <= 10; y += 0.2) {
	  for (let x = -10; x <= 10; x += 0.3) {
		let r = Math.sqrt(x * x + y * y) * 1.5;
		let f = r === 0 ? 90 : Math.abs(90 * Math.sin(r) / r);
		let a = Math.floor(10 * x + 160 - 5 * y);
		let b = Math.floor(5 * y + 2.5 * x + 93);
		point(a, b - f);
	  }
	}
}